lightning/src/lib.rs

```diff
@@ -28,31 +28,132 @@
 #![allow(bare_trait_objects)]
 #![allow(ellipsis_inclusive_range_patterns)]
 
+#![cfg_attr(all(not(feature = "std"), not(test)), no_std)]
+
 #![cfg_attr(all(any(test, feature = "_test_utils"), feature = "unstable"), feature(test))]
 #[cfg(all(any(test, feature = "_test_utils"), feature = "unstable"))] extern crate test;
 
+#[cfg(not(any(feature = "std", feature = "no_std")))]
+compile_error!("at least one of the `std` or `no_std` features must be enabled");
+
 #[macro_use]
 extern crate alloc;
 extern crate bitcoin;
+#[cfg(any(test, feature = "std"))]
 extern crate core;
+
 #[cfg(any(test, feature = "_test_utils"))] extern crate hex;
 #[cfg(any(test, feature = "fuzztarget", feature = "_test_utils"))] extern crate regex;
 
+#[cfg(feature = "no_std")] extern crate core2;
```

Comment:
I wonder if we should almost always use `not(feature = std)`, as, if both are set, we really just want to use `std::io` because it probably means someone should/will set the `core2/std` flag.

Reply:
Done